mlrun/runtimes/nuclio/function.py

```diff
@@ -655,6 +656,8 @@ def deploy(
         logger.info("Starting remote function deploy")
         data = db.deploy_nuclio_function(func=self, builder_env=builder_env)
         self.status = data["data"].get("status")
+        self.spec = merge(self.spec.to_dict(), data["data"].get("spec") or {})
```

Comment:
also, worth adding comment explaining what problem we are trying to solve here (smth about that we do .save() later and it may lead to rewriting values)